data/github.ts

```diff
@@ -0,0 +1,232 @@
+import dayjs from "dayjs";
+import { Issue as IssueModel, Repository as RepositoryModel } from "../types";
+
+import { Octokit } from "@octokit/core";
+import {
+  Issue,
+  IssueEdge,
+  Label,
+  LabelEdge,
+  Language,
+  Query,
+  Repository,
+  RepositoryTopic,
+  RepositoryTopicEdge,
+  SearchResultItemEdge,
+  validate
+} from "@octokit/graphql-schema";
+import { retry } from "@octokit/plugin-retry";
+import { throttling } from "@octokit/plugin-throttling";
+import { RequestOptions } from "@octokit/types";
+import millify from "millify";
+import slugify from "slugify";
+
+// Maximum number of issues to retrieve per repository. Only used by GitHub currently.
+const MAX_ISSUES = 10;
+
+// Setup Octokit (GitHub API client)
+const MyOctokit = Octokit.plugin(throttling, retry);
+const octokit = new MyOctokit({
+  auth: process.env.GH_PERSONAL_ACCESS_TOKEN,
+  throttle: {
+    onRateLimit: (retryAfter: number, options: object, octokit: Octokit, retryCount: number) => {
+      const { method, url } = options as RequestOptions;
+      octokit.log.warn(`Request quota exhausted for request ${method} ${url}`);
+
+      if (retryCount < 1) {
+        // only retries once
+        octokit.log.info(`Retrying after ${retryAfter} seconds!`);
+        return true;
+      }
+    },
+    onSecondaryRateLimit: (
+      retryAfter: number,
+      options: object,
+      octokit: Octokit,
+      retryCount: number
+    ) => {
+      const { method, url } = options as RequestOptions;
+      octokit.log.warn(`SecondaryRateLimit detected for request ${method} ${url}`);
+
+      if (retryCount < 2) {
+        // retries twice
+        octokit.log.warn(`Retrying after ${retryAfter} seconds!`);
+        return true;
+      }
+    }
+  }
+});
+
+/**
+ * Searches for GitHub repositories based on the provided search criteria.
+ *
+ * Uses {@link https://docs.github.com/en/graphql/overview/explorer GitHub's GraphQL API explorer} to
+ * build and test the search query.
+ *
+ * @param url - The URL of the GitHub instance to search.
+ * @param repositories - An array of repository names to search for.
+ * @param labels - An array of labels to filter issues by.
+ * @returns A Promise that resolves to an array of RepositoryModel objects.
+ * @throws An error if the GraphQL query is invalid.
+ */
+export const getGitHubRepositories = async (
```

Comment:
I suppose this could be called `getRepositories` now that `github` is in the name of the file. What do you think?

Reply:
I believe it would be best to keep it as it is at the moment as we will have to rename it in the import statement in the `shared.ts` file to prevent any conflicts. Currently, I do not see a significant advantage in doing so. Please let me know if you have a different opinion.